alacritty_terminal/src/clipboard.rs

```diff
@@ -12,53 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
-use std::ffi::c_void;
-
-use log::{debug, warn};
-
-use copypasta::nop_clipboard::NopClipboardContext;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
-use copypasta::wayland_clipboard;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
-use copypasta::x11_clipboard::{Primary as X11SelectionClipboard, X11ClipboardContext};
-use copypasta::{ClipboardContext, ClipboardProvider};
-
-pub struct Clipboard {
-    clipboard: Box<dyn ClipboardProvider>,
-    selection: Option<Box<dyn ClipboardProvider>>,
-}
-
-impl Clipboard {
-    #[cfg(any(target_os = "macos", target_os = "windows"))]
-    pub fn new() -> Self {
-        Self::default()
-    }
-
-    #[cfg(not(any(target_os = "macos", target_os = "windows")))]
-    pub fn new(display: Option<*mut c_void>) -> Self {
-        if let Some(display) = display {
-            let (selection, clipboard) =
-                unsafe { wayland_clipboard::create_clipboards_from_external(display) };
-            return Self { clipboard: Box::new(clipboard), selection: Some(Box::new(selection)) };
-        }
-
-        Self {
-            clipboard: Box::new(ClipboardContext::new().unwrap()),
-            selection: Some(Box::new(X11ClipboardContext::<X11SelectionClipboard>::new().unwrap())),
-        }
-    }
-
-    // Use for tests and ref-tests
-    pub fn new_nop() -> Self {
-        Self { clipboard: Box::new(NopClipboardContext::new().unwrap()), selection: None }
-    }
-}
-
-impl Default for Clipboard {
-    fn default() -> Self {
-        Self { clipboard: Box::new(ClipboardContext::new().unwrap()), selection: None }
-    }
+pub trait Clipboard: Send {
+    fn store(&mut self, ty: ClipboardType, text: String);
```

Comment:
Why did you change it from `Into<String>` to just `String`? If anything, it might even be more appropriate to take just `&str`.

Reply:
It was some issue with `trait` that forces me to do so, I feel like `&str` makes more sense though.